server/src/repositories/usersRepository.js

```diff
@@ -9,4 +9,59 @@ export const usersRepository = {
 
     return user;
   },
+
+  signup: async (email, github_oauth_id) => {
+    const connection = database.connection.getRepository(Users);
+    const savedUser = await connection.save({
+      email,
+      github_oauth_id,
+      github_oauth_token: "",
+    });
+  },
+
+  updateGithubOauthToken: async (github_oauth_id, github_oauth_token) => {
+    const connection = database.connection.getRepository(Users);
+    const user = await connection.findOne({
+      where: { github_oauth_id, github_oauth_id },
+    });
+
+    user.github_oauth_token = github_oauth_token;
+    const savedUser = await connection.save(user);
+  },
+
+  getUser: async (id) => {
+    const connection = database.connection.getRepository(Users);
+    return await connection.findOne({ where: { id: id } });
+  },
+
+  getUserByGithubId: async (github_oauth_id) => {
+    const connection = database.connection.getRepository(Users);
+    return await connection.findOne({
+      where: { github_oauth_id: github_oauth_id },
+    });
+  },
+
+  deleteUserByEmail: async (email) => {
+    const connection = database.connection.getRepository(Users);
+    const deleteResult = await connection.delete({ email: email });
+
+    if (deleteResult.affected == null || deleteResult.affected === 0) {
+      return false;
+    } else {
+      return true;
+    }
+  },
+
+  deleteUserByGithubId: async (github_oauth_id) => {
+    const connection = database.connection.getRepository(Users);
```

Comment:
```suggestion
    const connection = await database.connection.getRepository(Users);
```

await이 빠진게 아닌가 생각이 들어요

Reply:
위 코드에서는 connection 을 가져오는 작업이고, await 이 필요하지 않습니다. find 나 delete 로 db 에 접근할 때 await 이 필요합니다.

아마 일반적으로 connection 을 가져오는 코드와 이후 db 에 접근하는 코드(ex. find, delete) 을 한번에 쓰기 때문에 await 이 필요하다고 생각하신 것 같습니다.

일반적인 코드
```
await getConnection().getRepository(AnnualIncome).find();
```

현재 코드
```
const connection = database.connection.getRepository(Users);
const deleteResult = await connection.delete({
  github_oauth_id: github_oauth_id,
});
```

Reply:
아 그렇군요! 비동기 함수인줄 알았습니다 ㅎ